### ISSUE TITLE
Hyperparameters in cars generation

### ISSUE THREAD

**Issue by ming1993li** [christopher-beckham/hologan-pytorch#8]
Hi, great thanks for your work and it is amazing! I am reproducing the results of cars generation. I find some hyperparameters setting in "exp/example_cars.sh". So, are remaining parameters set as default values in "task_launcher.py"? Is that right? Besides, is this implementation just for resolution 64x64? If I need to generate images with dimension 128x128, I have to adjust the network architecture and some hyperparameters, like learning rate. Is that right？

**Comment by christopher-beckham** [comment]
Hi,

Yes, the rest are defaults in `task_launcher.py`. And yes, to have 128px generation, just add an extra strided resblock to the generator and likewise for the discriminator.

The two 'tricky' hyperparameters are `lamb` (the trade-off between the InfoGAN-style reconstruction loss and the GAN losses) and the `angles` one, which is set to `[0,0,-90,90,0,0]`, meaning: sample no rotation on the x-axis (i.e. the 'pitch'); sample rotations between -90 and 90 degrees on the y axis (i.e. the 'yaw' -- this is the upward axis in PyTorch); and sample no rotation on the z-axis (i.e. the 'roll'). Even though the cars dataset may have images of cars from a full 360 degrees of yaw, specifying `[...,-180,180,...]` may actually do something like more than one revolution of rotation. As far as I can best explain, this is simply because there is no guarantee that rotating by X degrees in _latent space_ corresponds to the same effect in _image space_, which is why I used ` [...,-90,90,...]` as my range.

Hope this helps!

**Comment by ming1993li** [comment]
Thank you very much for your detailed explanation. I was definitely confused about the yaw angle range. I will try it. Thank you again.
